Main/ClientApp/src/Components/Dialogs/ChangeMaxCollectionsDialog.tsx

```diff
@@ -21,33 +22,29 @@ export default function ChangeMaxCollectionsDialog(
     props.maxCollections
   );
 
+  useEffect(() => {
+    setMaxCollections(props.maxCollections);
+  }, [props.maxCollections]);
+
   return (
     <SimpleDialog
       open={props.open}
       toggleDialogOpen={props.toggleDialogOpen}
       title={title}
       description={description}
       content={
-        <TextField
-          onChange={(e) => setMaxCollections(e.target.value)}
-          autoFocus
-          margin="dense"
-          id="maxCollections"
-          label="Maximum number of collections"
-          type="number"
-          placeholder={"" + props.maxCollections}
-          defaultValue={"" + props.maxCollections}
-          fullWidth
+        <NumberInput
+          value={maxCollections}
+          onChange={(val) => setMaxCollections(val)}
         />
       }
       actions={
         <Button
           disabled={
-            maxCollections === props.maxCollections ||
-            ((maxCollections as unknown) as number) < 0
+            maxCollections === props.maxCollections || maxCollections <= 0
           }
           onClick={() => {
-            changeValue("MaxCollections", maxCollections);
+            props.onValueChange(maxCollections);
```

Comment:
i think that this component responsibility should be changing value of property "MaxCollections", so component should know what to do, but now the action is injected and this action may not change the property value.  if component would have been more abstract this approach would be good, but now it isn't

Reply:
The goal is to make it more abstract